### PR TITLE
Adding Helper To Mock Result Sets In Testing

### PR DESCRIPTION
**Pull request by Fenikkusu**
Hello!

* Type: new feature
* Link to issue: NA

This pull request affects the following components: **(please check boxes)**

* [ ] Library
* [ ] Code Style
* [ ] Documentation
* [x] Testing

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

I've added a trait for mocking a basic Result set. This comes in handy when doing unit tests and I don't want to have to actually connect to the DB to do so. 

Thanks

### CODE REVIEW

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Charles.Martin
+ * Date: 12/13/2016
+ * Time: 12:41 PM
+ */
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]: Could you please use proper DocBlock or remove this at all

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+namespace unit\Test\Traits;
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]: `unit\Test\Traits` => `Unit\Test\Traits`

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+use Phalcon\Test\Traits\ResultSet;
+use Codeception\TestCase\Test;
+
+class ResultSetTest extends Test
+{
+    use ResultSet;
+
+    /** @var \Phalcon\Test\Traits\ResultSet  */
+    protected $testSubject = NULL;
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]: `NULL` => `null`

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    public function setUp() {
+        $this->testSubject = $this;
+    }
+
+    public function testCanMockResultSet() {
+        $mockModel = $this->getMockBuilder('\Phalcon\Mvc\Model')
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]:
```php
use Phalcon\Mvc\Model

$mockModel = $this->getMockBuilder(Model:class)
```

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        /** @var \Phalcon\Mvc\Model\Resultset $mockResultSet */
+        $mockResultSet = $this->testSubject->mockResultset([]);
+
+        $this->assertEquals(0, $mockResultSet->count());
+        $this->assertFalse($mockResultSet->getFirst());
+        $this->assertFalse($mockResultSet->getLast());
+    }
+
+    public function testCanUseOtherResultSetClasses() {
+        $mockResultset = $this->mockResultset([], '\Phalcon\Mvc\Model\Resultset\Simple');
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]:
```php
Phalcon\Mvc\Model\Resultset\Simple;

$mockResultset = $this->mockResultset([], Simple::class);
```

File: tests/unit/Test/Traits/ResultSetTest.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        $this->assertSame($mockModel, $mockResultSet->getFirst());
+
+        //Testing getLast
+        $this->assertSame($mockThirdModel, $mockResultSet->getLast());
+
+        //Testing toArray
+        $this->assertSame($mockData, $mockResultSet->toArray());
+    }
+
+    public function testCanMockEmptyResultSet() {
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]: `public function testCanMockEmptyResultSet() {`

=> 

```
public function testCanMockEmptyResultSet()
{
```

File: tests/unit/Session/Adapter/AerospikeTest.php [hunk 7]
```diff
@@ -57,7 +57,9 @@ protected function _before()
      */
     protected function _after()
     {
-        $this->cleanup();
+        if (extension_loaded('aerospike')) {
+            $this->cleanup();
```
Comment (sergeyklay) [inline tests/unit/Session/Adapter/AerospikeTest.php]: 👍

File: Library/Phalcon/Test/Traits/ResultSet.php [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+ */
+trait ResultSet
+{
+    /**
+     * @param array $dataSet Mock Data Set To Use
+     * @param string $className ResultSet Class To Mimic (Defaults To Abstract ResultSet)
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject|\Phalcon\Mvc\Model\Resultset|\Phalcon\Mvc\Model\ResultsetInterface
+     */
+    public function mockResultSet(array $dataSet, $className = '\Phalcon\Mvc\Model\Resultset')
```
Comment (sergeyklay) [inline Library/Phalcon/Test/Traits/ResultSet.php]:
```php
use Phalcon\Mvc\Model\Resultset;

public function mockResultSet(array $dataSet, $className = Resultset::class)
{
```

File: tests/unit/Test/Traits/ResultSetTest.php (hunk 2 above)
Comment (Fenikkusu) [inline tests/unit/Test/Traits/ResultSetTest.php]: After double checking this (comparing other files), I believe this should actually be 'Phalcon\Test\Test\Traits'?

Comment (sergeyklay) [inline tests/unit/Test/Traits/ResultSetTest.php]: Yes

**Comment by Fenikkusu** [comment]
...Apparently I made the changes but never finished them or pushed them up...go figure...oops.

**Comment by sergeyklay** [comment]
@Fenikkusu Could you please rebase on 3.1.x branch

**Comment by sergeyklay** [comment]
The `3.0.x` branch is not longer supported. Could you please submit PR to the 3.2.x branch?

**Comment by Fenikkusu** [comment]
Updated, though it looks like the dump.sql may be out of date. Running Codeception locally is resulting in errors from EagerLoading and Prefixing.

**Comment by Fenikkusu** [comment]
Ignore my previous comment regarding the dump.sql - DB wasn't populated.

**Pull request merged**